src/components/Incident/IncidentSummary.js

```diff
@@ -0,0 +1,89 @@
+import React from 'react'
+import SyncIcon from 'material-ui/svg-icons/notification/sync'
+import IconButtonStyled from '../elements/IconButtonStyled'
+import ModeEditIcon from 'material-ui/svg-icons/editor/mode-edit'
+import Engagements from '../Engagements'
+import GlobalActions from '../Timeline/Playbook/GlobalActions'
+
+export const IncidentSummaryName = (ticketId) => {
+    return 'IncidentSummary' + (ticketId ? '_' + ticketId : '')
+}
+
+export const IncidentSummary = (incident, ticket, ticketSystem, ticketOriginId, dispatch) =>
+[
+    HeaderRow(ticketOriginId),
+    TicketDetailsRow(ticketSystem, ticket),
+    TitleRow(incident),
+    GlobalActionsRow(incident, ticketOriginId),
+    EngagementsRow(incident, dispatch)
+]
+
+const HeaderRow = (ticketId) => [
+    [
+        (key) =>
+            <strong key={key}>
+                Incident Summary{ticketId ? ` for ${ticketId}`:''}:
+                &nbsp;
+                <IconButtonStyled tooltip="Refresh">
+                    <SyncIcon />
+                </IconButtonStyled>
+            </strong>
+    ]
+]
+
+const TicketDetailsRow = (ticketSystem, ticket) => [
+    [
+        (key) =>
+            <a href={`${ticketSystem.ticketUriPrefix}${ticket.originId}${ticketSystem.ticketUriSuffix}`} key={key} target="_blank">
+                {ticket.originId}
+            </a>,
+        (key) =>
+            <div key={key}>
+                {ticket.severity}
+            </div>
+    ],
+    [
+        (key) =>
+            <div key={key}>
+                {ticket.imName}
+            </div>,
+        (key) =>
+            <IconButtonStyled tooltip="Edit IM" key={key}>
+                <ModeEditIcon />
+            </IconButtonStyled>
+    ]
+]
+
+const noTitleMessage = 'No Title!'
+
+const TitleRow = (incident) => [
+    <div>
+        {incident && incident.title
```

Comment:
The cascading logic here scares me. Can we use a `switch` statement instead?

Reply:
I'm not sure how to make a clean switch statement out of this since it can't really use strict comparison of values from incident. Extracted some of the multi-part validation to separate functions and changed this to a more simple form of ternary cascade (no branching)